pkg/client/rbac.go

```diff
@@ -0,0 +1,88 @@
+package client
+
+/*
+Usage example:
+
+type myhandler struct {
+	clientRbac client.Rbac
+}
```

Comment:
Since `pkg/client` seems to be specific to Rbac, I'm curious why not have this code in `pkg/middleware/rbac.go` instead?

Reply:
To elaborate, is this code in `pkg/client` only intended to be used with the middleware code, or is it something you think could be used independently?

Reply:
1st comment
- I thought that a logical layer to break down is the client code to interact with third party platform services.
- I could have created a new file `pkg/middleware/rbac_client.go`, but I thought about it in the future as a common group of components with a common responsibility (the line above).
- The above line align with future feature integrating pulp service, so it would provide a place into the repository to hold that at `pkg/client`.
- If in the future we have an additional interaction for other platform service, we only had to add the code to the `pkg/client` package.

2nd comment

- I think it will be used only with the middleware code, but I see two logical responsibilities:
  - the middleware provide a centralised place to check the permissions.
  - the client provide the communication with the third party component (in this case it is only an adapter to the public client library). If it were generated from the rbac api, it would have generated a complete component to interact with the whole rbac api.
- I don't think it will be used independently.